### ISSUE TITLE
TUN interface is not created

### ISSUE THREAD

**Issue by SharadaJay** [Orange-OpenSource/towards5gs-helm#46]
Hi,

I'm currently working on helm installation of free5gc and I'm using towards5gs-helm. I can make all the pods running but unfortunately the TUN interface is not created for me. I tried to troubleshoot it but haven't found a solution yet . I have attached the screenshots of ue, gNB, AMF, SMF and UPF logs for your kind reference.

Best Regards,
Sharada.

![image](https://user-images.githubusercontent.com/66205107/177239171-8d4174da-b291-459f-afff-15ec03871c33.png)
![image](https://user-images.githubusercontent.com/66205107/177239318-dca56146-9b53-4617-bca1-3beb2734fbb7.png)
![image](https://user-images.githubusercontent.com/66205107/177239701-315a700b-dce7-4d3b-8214-4becac116ea1.png)
![image](https://user-images.githubusercontent.com/66205107/177239841-407364c4-11b3-4ecb-8884-9c3d80eb9745.png)
![image](https://user-images.githubusercontent.com/66205107/177239907-4747187a-e0bf-48a6-b540-9fc3013f8a25.png)
![image](https://user-images.githubusercontent.com/66205107/177239976-341045f9-a95c-4304-97dc-1e5feebcb167.png)
![image](https://user-images.githubusercontent.com/66205107/177240161-81010027-6055-4710-ac15-1b181ea843e4.png)
![image](https://user-images.githubusercontent.com/66205107/177240377-c430b79d-2140-4ea5-b717-ca142b734125.png)
![image](https://user-images.githubusercontent.com/66205107/177241407-5f4a6fd4-5e94-44ab-97a1-c6675186ecfc.png)
![image](https://user-images.githubusercontent.com/66205107/177241472-dee7336c-4658-4f8d-ad26-4361fae5bbaa.png)
![image](https://user-images.githubusercontent.com/66205107/177241552-4331ee2e-5a01-42e9-907e-eb8cb8d8d652.png)

**Comment by raoufkh** [comment]
@konradkar2 is right. The gNB cannot connect to the AMF. Are you waiting for the AMF Pod to be up before installing UERANSIM?

**Comment by SharadaJay** [comment]
Hi @konradkar2 @raoufkh,

Thank you for the reply. 

@raoufkh yes I installed UERANSIM after the AMF pod is up and running. I can ping to 10.100.50.249 from the gNB pod but cannot telnet to the 38412 port. The amf pod is only listening on port 80 and does not listen on port 38412. Also I noticed that a service is not created for SCTP.

![image](https://user-images.githubusercontent.com/66205107/178900727-6235e7e2-6eb1-458d-868c-bb3a4a4a7919.png)
![image](https://user-images.githubusercontent.com/66205107/178900727-6235e7e2-6eb1-458d-868c-bb3a4a4a7919.png)
![image](https://user-images.githubusercontent.com/66205107/178901854-5b650089-7046-4210-8810-8bb748182590.png)
![image](https://user-images.githubusercontent.com/66205107/178901854-5b650089-7046-4210-8810-8bb748182590.png)
![image](https://user-images.githubusercontent.com/66205107/178902853-1c78e974-2e0a-45ef-ab50-4ad6c10572ca.png)
![image](https://user-images.githubusercontent.com/66205107/178902853-1c78e974-2e0a-45ef-ab50-4ad6c10572ca.png)
![image](https://user-images.githubusercontent.com/66205107/178903636-ea5c6bdc-c6d8-49ff-8bdf-85aea969ac75.png)
![image](https://user-images.githubusercontent.com/66205107/178903636-ea5c6bdc-c6d8-49ff-8bdf-85aea969ac75.png)

**Comment by raoufkh** [comment]
AMF logs show that AMF is listening on 10.100.50.249:38412. Can you show the network interfaces that are created in the gNB pod?

**Comment by oanabutaru98** [comment]
Hello! I am having a similar issue when trying to deploy the helms. The TUN interface is not created on the UE, however, I have the following error in the AMF logs.

2023-02-17T12:04:34Z [INFO][AMF][NGAP][10.100.50.250:41800] Handle Initial UE Message
2023-02-17T12:04:34Z [INFO][LIB][FSM] Handle event[Gmm Message], transition from [Deregistered] to [Deregistered]
2023-02-17T12:04:34Z [INFO][AMF][GMM][AMF_UE_NGAP_ID:34] Handle Registration Request
2023-02-17T12:04:34Z [INFO][LIB][FSM] Handle event[Start Authentication], transition from [Deregistered] to [Authentication]
2023-02-17T12:04:34Z [INFO][AMF][GMM][AMF_UE_NGAP_ID:34] Authentication procedure
2023-02-17T12:04:34Z [ERRO][AMF][GMM][AMF_UE_NGAP_ID:34] Nausf_UEAU Authenticate Request Failed: &{Type: Title: Status:500 Detail: Instance: Cause:AV_GENERATION_PROBLEM InvalidParams:[]}

I would appreciate some advice, thank you!

**Comment by oanabutaru98** [comment]
you are right, this seems to solve the problem for me as well

**Comment by lgabhishek18** [comment]
Hi,
I am also facing the same issue. My all pods running but unfortunately the TUN interface is not created.

Just to summarize, I have created a Kubernetes cluster (with 2 nodes) using kubeadm to deploy free5gc and ueransim.
Detailed info:
Ubuntu: 20.04.6 LTS
Kernel version: 5.4.0-152-generic
Container Runtime version: containerd://1.6.12
K8s Version: v1.20.1
Free5gc: v3.3.0
Ueransim: v3.2.6

![image](https://github.com/Orange-OpenSource/towards5gs-helm/assets/60277134/b88fedba-7e7e-4cbf-8485-118fe66d6025)

![image](https://github.com/Orange-OpenSource/towards5gs-helm/assets/60277134/9f974fb2-f896-49c2-919d-78c157ac7785)

![image](https://github.com/Orange-OpenSource/towards5gs-helm/assets/60277134/60906fe7-0601-49bb-a410-7f5df431b751)

(Note: I have deployed ueransim after my all pods of free5gc are running and I have manually created a New Subscriber from web interface also)

**Ueransim logs:**
![image](https://github.com/Orange-OpenSource/towards5gs-helm/assets/60277134/6be2b0f8-fb77-45ea-9434-a76b1fd86dbe)
**Network Interfaces availabe in gNB:**
![image](https://github.com/Orange-OpenSource/towards5gs-helm/assets/60277134/94fe0acc-0ff6-46a1-961f-4c7d98e95bde)
**Network Interfaces availabe in ue:**
![image](https://github.com/Orange-OpenSource/towards5gs-helm/assets/60277134/0c930f6f-e021-4d68-87e3-ec86aa042a3a)

**AMF logs:**
![image](https://github.com/Orange-OpenSource/towards5gs-helm/assets/60277134/3cd0bd39-2359-43de-be76-f87e4c61a531)

If other logs are required please do let me know.

I'm looking forward to receive some suggestion on this, Thanks!

**Comment by ssdumbre** [comment]
![image](https://github.com/Orange-OpenSource/towards5gs-helm/assets/43669080/e8abcab5-e557-4a18-b160-d7583870a494)
![image](https://github.com/Orange-OpenSource/towards5gs-helm/assets/43669080/be4e112b-56ae-49b2-ae89-aae20585edfd)
![image](https://github.com/Orange-OpenSource/towards5gs-helm/assets/43669080/7463f744-06bc-4693-a8ee-ca7668b91d50)
I am also facing similar issue , please help here

**Comment by ssdumbre** [comment]
I have tried the suggested workaround but it did not work for me.

**Comment by gondormartin** [comment]
I faced the same problem. Hope these to be useful:
I found this when I am trying to install uneransim's chart:
![image](https://github.com/Orange-OpenSource/towards5gs-helm/assets/144603473/0ac73e7c-387f-4c73-ac29-7eb2ba06bc44)
So,I uninstalled the uneransim, then I created a subcriber in UI as @arielgoes mention above.
After this, I installed uneransim's chart again, then pods ue and gnb started.
Now a UE is registerd, we can see it in free5gc's web UI page: 
![image](https://github.com/Orange-OpenSource/towards5gs-helm/assets/144603473/4d48e4bf-2edd-41a9-b0fb-a43ebb3b8cb7)
Then I enter the pod and finally see the TUN
![image](https://github.com/Orange-OpenSource/towards5gs-helm/assets/144603473/557ca9f1-bb10-4beb-8b3a-4174003ab423)
